File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        String w = s.toLowerCase().replaceAll("[^A-Za-z0-9]","");
        if(w.length()==1) return true;
        for(int i=0;i<w.length()/2;i++){
           if(w.charAt(i) != w.charAt(w.length()-1-i)) return false;
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def is_alph(self, c):
        return (ord('A') <= ord(c) <= ord('Z') or 
                ord('a') <= ord(c) <= ord('z') or 
                ord('0') <= ord(c) <= ord('9'))

    def isPalindrome(self, s: str) -> bool:
        s = s.lower()
        
        l, r = 0, len(s) - 1
        while l < r:
            while l < r and not self.is_alph(s[l]):
                l += 1
            while l < r and not self.is_alph(s[r]):
                r -= 1
            if s[l] != s[r]:
                return False
            l, r = l + 1, r - 1

        return True